docs/source/lesson/prg/python_intro.rst

```diff
@@ -261,27 +121,55 @@ At the end of this lesson you will be able to:
 - have an overview of the standard library
 - install Python libraries using ``virtualenv``
 
+Python Installation
+=================================================================
 
-.. _python-resources:
+Python is easy to install and very good instructions for most
+platforms can be found on the python.org Web page. We will be using
+Python 2.7.12 but not Python 3.
 
-Using Python on FutureSystems
-=================================================================
+In addition to Python, it is useful to have `pip <>`_ package
+installation tool on your system.
 
-.. warning:: This is only important if you use Futuresystems resources.
+In the tutorial, we assume that you have a computer with python
+installed.  However, we also recommend that forthe class you use
+Python's virtualenv (see below) to isolate your development Python
+from the system installed Python.
 
-In order to use Python you must log into your FutureSystems account.
-Then at the shell prompt execute the following command::
+.. _virtualenv_:
 
-  $ module load python
+virtualenv
+------------------------------------------------------
 
-This will make the ``python`` and ``virtualenv`` commands available to
-you.
+Often you have your own computer and you do not like to change its
+environment to keep it in prestine condition. Python comes with mnay
+libraries that could for example conflict with libraries that you have
+installed. To avoid this it is bets to work in an isolated python
+environment while using virtualenv,. Documentation about it can be
+found at::
 
+* https://virtualenv.pypa.io
 
-.. tip::
+The installation is simple once you have pip installed. If it is not
+installed you can say::
 
-   The details of what the ``module load`` command does are described
-   in the future lesson :doc:`modules`.
+  $ easy_install pip
+
+After that you can install the virtual env with::
+
+  $ pip install virtualenv
+
+To setup an isolated environment for example in the directory ~/ENV
+please use::
+
+  $ virtualenv ~/ENV
+
+To activate it you can use the command::
+
+  $ source ~/ENV/bin/activate
+
+you can put this command n your bashrc or bash_profile command so you
+do not forget to activate it.
```

Comment:
Perhaps link to a linux lesson where bash and shell init files are discussed. I don't think such a lesson exists right now, but you can create a stub with a warning, not implemented yet message